furryfeet_be/furryfeet/views.py

```diff
@@ -80,6 +82,57 @@ def post(self, request):
         
         return response
 
+class ChangeUserPasswordView(APIView):
+    def post(self, request):
+        token = request.COOKIES.get('jwt')
+        
+        if not token:
+            raise AuthenticationFailed("Unauthenticated!")
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed("Unauthenticated!")
+        
+        user = User.objects.get(id=payload['id'])
+        current_password = request.data["current_password"]
+        new_password1 = request.data["new_password1"]
+        new_password2 = request.data["new_password2"]
+        
+        # Verify the current password
+        if not user.check_password(current_password):
+            
+            return Response({"error": "Incorrect current password"}, status=400)
+        
+        # Verify that the new passwords match
+        if new_password1 != new_password2:
+            return Response({"error": "New passwords do not match"}, status=400)
+        
+        # Verify that the new password meets the policy requirements
+        if not is_password_valid(new_password1):
+            return Response({"error": "New password does not meet the policy requirements"}, status=400)
+        
+        # Change the user's password and update the session
+        user.set_password(new_password1)
+        user.save()
+        update_session_auth_hash(request, user)
+        
+        return Response({"message": "Password successfully changed"})
+
+def is_password_valid(password):
```

Comment:
Probably it would be better to extract this func in a different script as it is not a controller by itself